client/components/common/SelectElement/index.vue

```diff
@@ -2,88 +2,174 @@
-    allowedTypes: { type: Array, default: () => [] }
+    submitLabel: { type: String, default: 'Save' },
+    headerIcon: { type: String, default: 'mdi-toy-brick-plus-outline' },
+    useCurrentRepo: { type: Boolean, default: false }
   },
   data: () => ({
+    repository: null,
     selectedActivity: null,
     contentContainers: [],
     selectedElements: [],
-    loadingContent: false
+    loadingContent: false,
+    activities: []
   }),
-  computed: mapGetters('repository', ['repository', 'activities']),
+  computed: {
+    ...mapGetters('repository', {
+      currentRepository: 'repository',
+      currentActivities: 'activities'
+    }),
+    allElementsSelected: vm => vm.selectedElements.length === vm.elements.length,
+    showBackButton: vm => vm.useCurrentRepo ? !!vm.selectedActivity : !!vm.repository,
+    processedContainers() {
+      const { selectedActivity, activities } = this;
+      if (!selectedActivity || !activities.length) return [];
+      const rootTypes = getContainerTypes(selectedActivity.type);
+      let containers = activities.filter(({ type, parentId }) => {
+        return parentId === selectedActivity.id && rootTypes.includes(type);
+      });
+      containers = sortBy(containers, [
+        it => rootTypes.indexOf(it.type), 'position', 'createdAt'
+      ]);
+      return containers.reduce((acc, container) => {
+        const subcontainers = getContainers(activities, container);
+        acc.push(container, ...sortBy(subcontainers, 'position'));
+        return acc;
+      }, []);
+    },
+    elements() {
+      const elements = flatMap(this.contentContainers, 'elements');
+      if (!this.allowedTypes.length) return elements;
+      return elements.filter(it => this.allowedTypes.includes(it.type));
+    },
+    toggleButton() {
+      const { allElementsSelected, elements, multiple, selectedActivity } = this;
+      if (!multiple || !selectedActivity || !elements.length) return;
+      const { SELECT, DESELECT } = TOGGLE_BUTTON;
+      return allElementsSelected ? DESELECT : SELECT;
+    }
+  },
   methods: {
     async showActivityElements(activity) {
       this.selectedActivity = activity;
-      const { activities } = this;
-      const containers = sortBy(getContainers(activities, activity), 'position');
-      const elements = await this.fetchElements(containers);
-      this.contentContainers = containers.map(container => ({
-        ...container,
-        elements: elements.filter(element => element.activityId === container.id)
-      }));
+      const { processedContainers } = this;
+      const elements = await this.fetchElements(processedContainers);
+      this.contentContainers = processedContainers.map(container => {
+        const containerElements = elements
+          .filter(it => it.activityId === container.id)
+          .map(element => ({ ...element, activity }));
+        return { ...container, elements: sortBy(containerElements, 'position') };
+      });
```

Comment:
It might be worth to abstract adding elements to a container here:
```suggestion
      this.contentContainers = processedContainers.map(this.addContainerElements);
```

Reply:
Some more context variables needed (`activity`, `elements`) so please take a look if you find new solution more readable 😉 .